### ISSUE TITLE
aws_elasticache_replication_group redis AZ list must match node number on create but not update

### ISSUE THREAD

**Issue by nomeelnoj** [hashicorp/terraform-provider-aws#14070]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

`v0.12.24`
### Affected Resource(s)

* aws_elasticache_replication_group

### Terraform Configuration Files

```hcl
resource "aws_elasticache_replication_group" "this" {
  replication_group_id          = var.cluster_id
  replication_group_description = "Redis cluster ${var.cluster_id}"
  number_cache_clusters         = 4
  engine_version                = "5.0.6"
  subnet_group_name             = aws_elasticache_subnet_group.elasticache_redis_subnet_group.name
  security_group_ids            = concat([aws_security_group.redis_security_group.id], var.security_group_ids)
  apply_immediately             = true
  node_type                     = "cache.m5.large"
  automatic_failover_enabled    = true
  availability_zones            = ["us-east-1a", "us-east-1b", "us-east-1c"]
  engine                        = "redis"
  at_rest_encryption_enabled    = true
  transit_encryption_enabled    = true
  parameter_group_name          = "default.redis5.0"
  port                          = var.port
  tags                          = var.tags
}
```
### Expected Behavior

When setting `availability_zones` to a different number than `number_cache_clusters`, all nodes should be created in those availability zones.

### Actual Behavior

An error is thrown: 
```
Error: Error creating Elasticache Replication Group: InvalidParameterValue: When specifying preferred availability zones, the number of cache clusters must be specified and must match the number of preferred availability zones.
	status code: 400, request id: b7f8cc2b-ce91-452a-8760-4b8b605bdfe5
```
However, if you create initially with a matching number and then go update the number of nodes, terraform does not throw the error and you are able to create additional nodes without providing additional AZs.  Same is true for removing nodes.

### Steps to Reproduce

1. Create a config for a replication group
2. Provide `redis` as the engine type
3. Set the list of AZs to a diff number than `number_cache_clusters`
4. `terraform apply`
5. Change the configs so the number of cache clusters and azs match
6. `terraform apply`
7. Modify the value and add another cache cluster
8. `terraform apply`

### References

* #207 looks like it fixed this for memcached, but the issue persists for redis.

**Comment by isurewould** [comment]
Experiencing this as well!

**Comment by oguzhanaygn** [comment]
I'm facing the exact same issue here.

**Comment by gdavison** [comment]
Thank you for submitting this issue, @nomeelnoj. Fixing this bug will require a breaking change to the resource. Typically, we only make breaking changes to the provider with major version releases. However, when it comes to a bug, we sometimes make an exception.

The `availability_zones` parameter was unfortunately declared as a `TypeSet ` where the order does not matter and there can be no duplicates, while in the AWS API the order does matter and duplicates can exist. I believe that this was due to a misunderstanding of how the API uses `availability_zones` when the resource was first coded. The parameter does not specify which availability zones are valid, it specifies which availability zone each cache cluster is deployed in, the primary node being deployed in the first AZ, and the replicas in the remainder.

In your case, you appear to be deploying the Replication Group in a VPC, so the  ElastiCache Subnet Group indicated by `subnet_group_name` is enough to specify which AZs to use. You don't need to use `availability_zones` unless you want specific cache clusters in specific AZs, but since it's currently implemented as a `TypeSet` the order can not be controlled.

We will make a decision about how to best address the error.

**Comment by gdavison** [comment]
Closed by #22925
